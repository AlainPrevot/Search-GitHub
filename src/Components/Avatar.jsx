import StyleAvatar from './sass/Avatar.module.sass'
import PropTypes from 'prop-types'

export const Avatar = ({avatar_url}) => {
 
  return (
    <img className={StyleAvatar.avatar} src={avatar_url} alt="Avatar" />
  )
}

Avatar.propTypes = { 
  avatar_url: PropTypes.string.isRequired,
}