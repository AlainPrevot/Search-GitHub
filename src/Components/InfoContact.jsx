import {IconCompany, IconLocation, IconTwitter, IconURL} from '../Icons'
import StyleInfoContact from './sass/InfoContact.module.sass'
import PropTypes from 'prop-types'

export const InfoContact = ({location, twitter, url, company}) => {

    let user_location = location == null ? 'Not Available' : location
    let user_twitter = twitter == null ? 'Not Available' : twitter
    let user_url = url == null ? 'Not Available' : url
    let user_company = company == null ? 'Not Available' : company

  return (
    <div className={StyleInfoContact.content}>

        <IconLocation className={StyleInfoContact.content__Icon}/>
        <span>{user_location}</span>

        <IconTwitter className={StyleInfoContact.content__Icon}/>
        <span>{user_twitter}</span>

        <IconURL className={StyleInfoContact.content__Icon}/>
        <a href={user_url}>{user_url}</a>

        <IconCompany className={StyleInfoContact.content__Icon}/>
        <span>{user_company}</span>
    </div>
  )
}

InfoContact.propTypes = {
  location: PropTypes.string.isRequired,
  twitter: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  company: PropTypes.string.isRequired,
}
