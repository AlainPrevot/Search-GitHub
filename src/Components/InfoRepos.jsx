import StyleInfoRepos from './sass/InfoRepos.module.sass'

export const InfoRepos = ({repos=0, followers=0, following=0}) => {

  return (
    <div className={StyleInfoRepos.content}>

      <div>
        <span>Repos</span>
        <span>{repos}</span>
      </div>

      <div>
        <span>Folowers</span>
        <span>{followers}</span>
      </div>
      
      <div>
        <span>Following</span>
        <span>{following}</span>
      </div>
    </div>
  )
};

