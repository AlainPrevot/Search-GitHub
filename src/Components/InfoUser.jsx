import { useDate } from '../hooks/useDate'
import StyleInfoUser from './sass/InfoUser.module.sass'

export const InfoUser = ({user, name, username, bio}) => {

    let biography = bio == null ? 'This profile has no bio' : bio

    const {day, month, year} = useDate(user);
  return (
    <div className={StyleInfoUser.content}>
        <h1>{name}</h1>
        <h2>{username}</h2>
        <p>Joined {day} {month} {year}</p>  
        <p>{biography}</p>
    </div>
  )
}
