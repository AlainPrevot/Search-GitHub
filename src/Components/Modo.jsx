import StyleModo from './sass/Modo.module.sass'

export const Modo = ({onMode}) => {
  
  return (
    <div className={StyleModo.content}>

        <h1>devfinder</h1>

        <div className={StyleModo.modo}>

            <input id="modo" type="checkbox" name='my-checkbox' onClick={() => onMode()}/>
            <label className={StyleModo.dark} htmlFor="modo">DARK</label>
            <label className={StyleModo.ligth} htmlFor="modo">LIGTH</label>
        </div>
    </div>
  )
}
