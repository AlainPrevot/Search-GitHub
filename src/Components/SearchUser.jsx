import { useState } from "react"
import { IconLupa } from "../Icons";
import StyleSearchUser from "./sass/SearchUser.module.sass";
import PropTypes from 'prop-types'

export const SearchUser = ({onNewUser}) => {

    const [inputValue, setInputValue] = useState(''); //Creo un hook

    const onInputChange =  e => { {/* Esta funcion envia los datos al input cada vez que escribo */}
        setInputValue(e.target.value); //Imprimo lo qu escribo en el input
    };

    const onSubmit = e => { //Esta funcion se ejecuta cada vez que presionamos enter
        e.preventDefault(); //Hacemos que no se recargue la pagina
        if(inputValue.trim().length <= 1) return //Controlamos que el input no este vacio y eliminamos los espacios al principio y al final 

        onNewUser(inputValue.trim()); //Enviamos el resultado fuera del componente
        setInputValue(''); //Limpiamos el componente
    };


  return (
    <form className={StyleSearchUser.form} onSubmit={onSubmit}  aria-label='form'> {/* La funcion OnSubmit hace que no se recargue la pagina cada vez que buscamos un valor */}

        <IconLupa className={StyleSearchUser.form__lupa}/>

        <input 
            className={StyleSearchUser.input}
            type="text" 
            placeholder="Search GitHub username"
            value={inputValue}
            onChange={onInputChange}
        />
        
        <button className={StyleSearchUser.button} onSubmit={onSubmit}>Search</button>
    </form>
  )
};

SearchUser.propTypes = {
    onNewUser: PropTypes.func.isRequired,
}
