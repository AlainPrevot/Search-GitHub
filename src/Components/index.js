export * from './Avatar';
export * from './CardUser';
export * from './InfoContact';
export * from './InfoRepos';
export * from './InfoUser';
export * from './Modo';
export * from './SearchUser';

