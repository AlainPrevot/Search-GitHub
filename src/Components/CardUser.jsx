import { Avatar } from "./Avatar";
import { useFetchGitHub } from "./hooks/useFetchGitHub"
import { InfoContact } from "./InfoContact";
import { InfoRepos } from "./InfoRepos";
import { InfoUser } from "./InfoUser";

import './sass/CardUser.sass'

export const CardUser = ({username}) => {


  const {avatar} = useFetchGitHub(username);


  return (
    <div className="card">
        <Avatar 
          avatar_url={avatar.avatar_url}
        />
        <InfoUser
          name={avatar.name}
          username={avatar.login}
          bio={avatar.bio}
          created_at={avatar.created_at}
        />
        <InfoRepos
          repos={avatar.public_repos}
          followers={avatar.followers}
          following={avatar.following}
        />
        <InfoContact
          location={avatar.location}
          twitter={avatar.twitter_username}
          url={avatar.html_url}
          company={avatar.company}
        />
    </div>
  )
}
