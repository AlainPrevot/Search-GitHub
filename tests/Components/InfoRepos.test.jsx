import { render, screen } from "@testing-library/react";
import { InfoRepos } from "../../src/Components/InfoRepos";

describe('Pruebas en InfoRepos', () => { 
    
    test('Debe de hacer match con el snapshot', () => { 
     
        const {container} = render(<InfoRepos repos={8} followers={22} following={10}/>);
        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar las propiedades pasadas', () => {

        render(<InfoRepos repos={8} followers={22} following={10}/>);
        expect(screen.getByText(8)).toBeTruthy();
        expect(screen.getByText(22)).toBeTruthy();
        expect(screen.getByText(10)).toBeTruthy();

    });

 });