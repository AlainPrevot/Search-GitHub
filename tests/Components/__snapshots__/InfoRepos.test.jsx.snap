// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en InfoRepos Debe de hacer match con el snapshot 1`] = `
<div>
  <div>
    <div>
      <span>
        Repos
      </span>
      <span>
        8
      </span>
    </div>
    <div>
      <span>
        Folowers
      </span>
      <span>
        22
      </span>
    </div>
    <div>
      <span>
        Following
      </span>
      <span>
        10
      </span>
    </div>
  </div>
</div>
`;
